mm2src/coins/utxo/utxo_balance_events.rs

```diff
@@ -58,58 +58,35 @@ impl EventStreamer for UtxoBalanceEventStreamer {
         const RECEIVER_DROPPED_MSG: &str = "Receiver is dropped, which should never happen.";
         let streamer_id = self.streamer_id();
         let coin = self.coin;
+        let mut scripthash_to_address_map = HashMap::new();
 
-        async fn subscribe_to_addresses(
-            utxo: &UtxoCoinFields,
-            addresses: HashSet<Address>,
-        ) -> Result<BTreeMap<String, Address>, String> {
-            match utxo.rpc_client.clone() {
-                UtxoRpcClientEnum::Electrum(client) => {
-                    // Collect the scrpithash for every address into a map.
-                    let scripthash_to_address_map = addresses
-                        .into_iter()
-                        .map(|address| {
-                            let scripthash = address_to_scripthash(&address).map_err(|e| e.to_string())?;
-                            Ok((scripthash, address))
-                        })
-                        .collect::<Result<HashMap<String, Address>, String>>()?;
-                    // Add these subscriptions to the connection manager. It will choose whatever connections
-                    // it sees fit to subscribe each of these addresses to.
-                    client
-                        .connection_manager
-                        .add_subscriptions(&scripthash_to_address_map)
-                        .await;
-                    // Convert the hashmap back to btreemap.
-                    Ok(scripthash_to_address_map.into_iter().map(|(k, v)| (k, v)).collect())
-                },
-                UtxoRpcClientEnum::Native(_) => {
-                    Err("Balance streaming is currently not supported for native client.".to_owned())
-                },
-            }
-        }
-
-        if coin.as_ref().rpc_client.is_native() {
-            let msg = "Native RPC client is not supported for UtxoBalanceEventStreamer.";
+        // Make sure the RPC client is not native. That doesn't support balance streaming.
+        if let UtxoRpcClientEnum::Native(_) = coin.as_ref().rpc_client {
```

Comment:
You can use `if coin.as_ref().rpc_client.is_native()` as was done before

Reply:
[bb2b846](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2431/commits/bb2b8465bf4c0f166d2309cfbd082bc0b95f8b38)

Reply:
[bb2b846](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2431/commits/bb2b8465bf4c0f166d2309cfbd082bc0b95f8b38)